pkg/source/reconciler/mtsource/kafkasource.go

```diff
@@ -120,14 +120,14 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, src *v1beta1.KafkaSource
 	defer c.Close()
 	src.Status.MarkConnectionEstablished()
 
-	err = client.InitOffsets(ctx, c, src.Spec.Topics, src.Spec.ConsumerGroup)
+	err, totalPartitions := client.InitOffsets(ctx, c, src.Spec.Topics, src.Spec.ConsumerGroup)
 	if err != nil {
 		logging.FromContext(ctx).Errorw("unable to initialize consumergroup offsets", zap.Error(err))
 		src.Status.MarkInitialOffsetNotCommitted("OffsetsNotCommitted", "Unable to initialize consumergroup offsets: %v", err)
 		return err
 	}
 	src.Status.MarkInitialOffsetCommitted()
-
+	src.Status.MaxAllowedVReplicas = totalPartitions*r.scheduler.GetMaxEventPerSecondPerPartition()/r.scheduler.GetVReplicaMPS() + 1
```

Comment:
only the Kafka controller should know about MaxEventPerSecondPerPartition and VReplicaMPS, not the scheduler.

Reply:
ok will change